File: fintrack-backend/src/main/java/com/fintrack/fintrackbackend/model/Transaction.java
package com.fintrack.fintrackbackend.model;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(length = 255)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TransactionType type;

    @Column(name = "category_id", nullable = false)
    private Integer categoryId;

    @Column(name = "goal_id")
    private Integer goalId;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        this.date = LocalDate.now();
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public LocalDate getDate() { return date; }
    public void setDate(LocalDate date) { this.date = date; }

    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public TransactionType getType() { return type; }
    public void setType(TransactionType type) { this.type = type; }

    public Integer getCategoryId() { return categoryId; }
    public void setCategoryId(Integer categoryId) { this.categoryId = categoryId; }

    public Integer getGoalId() { return goalId; }
    public void setGoalId(Integer goalId) { this.goalId = goalId; }

    public Integer getUserId() { return userId; }
    public void setUserId(Integer userId) { this.userId = userId; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: frontend/app/src/main/java/pe/fintrack/mobile/ui/theme/data/Transaction.kt
package pe.fintrack.mobile.ui.theme.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.math.BigDecimal

@JsonClass(generateAdapter = true)
data class Transaction(
    @Json(name = "id") val id: Long,
    @Json(name = "date") val date: String,
    @Json(name = "amount") val amount: BigDecimal,
    @Json(name = "description") val description: String?,
    @Json(name = "type") val type: TransactionType,
    @Json(name = "categoryId") val categoryId: Int,
    @Json(name = "goalId") val goalId: Long?,
    @Json(name = "userId") val userId: Int,
    @Json(name = "createdAt") val createdAt: String
)

enum class TransactionType {
    INCOME,
    EXPENSE
}
